weatherreportntest.cpp

```diff
@@ -0,0 +1,69 @@
+#include <cassert>
+#include <iostream>
+#include <vector>
+#include <string>
+#include "./weatherreportn.h"
+
+using std::string;
+using std::vector;
+
+class SensorStub : public WeatherSpace::IWeatherSensor {
+ public:
+    int Humidity() const override {
+        return 72;
+    }
+
+    int Precipitation() const override {
+        return 70;
+    }
+
+    double TemperatureInC() const override {
+        return 26;
+    }
+
+    int WindSpeedKMPH() const override {
+        return 52;
+    }
+};
+
+// New stub for testing different scenarios
+class HighPrecipitationLowWindStub : public WeatherSpace::IWeatherSensor {

```

Comment:
Making a different class for a different scenario definitely serves the purpose.
Another choice would be: `SensorStub` is a class already, and it's possible to have multiple instances (objects) of that class. If you could pass the desired values in the constructor, each object could remember and return those values. Then the names (`highPrecipitationLowWindStub`) could be given to the instance variables. How does this sound?